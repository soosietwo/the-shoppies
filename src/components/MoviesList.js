import React from "react";
import { ResourceList } from "@shopify/polaris";

import EmptyState from "./EmptyState";

const MoviesList = (props) => {
  const {
    movies,
    totalResults,
    singular,
    plural,
    renderItem,
    emptyStateTitle,
    emptyStateCaption,
  } = props;

  return (
    <ResourceList
      emptyState={
        <EmptyState
          movies={movies}
          emptyStateTitle={emptyStateTitle}
          emptyStateCaption={emptyStateCaption}
        />
      }
      resourceName={{ singular, plural }}
      items={movies}
      renderItem={(movie) => renderItem(movie)}
      totalItemsCount={totalResults}
    />
  );
};

export default MoviesList;
