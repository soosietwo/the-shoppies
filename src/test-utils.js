// test-utils.js
import React from "react";
import { render } from "@testing-library/react";
import { AppProvider, Frame } from "@shopify/polaris";
import en from "@shopify/polaris/locales/en.json";
import Store from "./store";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const AllTheProviders = ({ children }) => {
  return (
    <AppProvider i18n={en}>
      <Store>
        <Frame>{children}</Frame>
      </Store>
    </AppProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
