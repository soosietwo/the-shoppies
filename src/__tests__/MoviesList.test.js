import React from "react";
import { render } from "../test-utils";
import MoviesList from "../components/MoviesList";

describe("MoviesList", () => {
  it("renders and matches snapshot", () => {
    const { container } = render(
      <MoviesList
        movies={[{ title: "Test" }]}
        totalResults={1}
        singular="movie"
        plural="movies"
        renderItem={(movie) => <div>{movie.title}</div>}
        emptyStateTitle="Test empty state"
        emptyStateCaption="Test empty state caption"
      />
    );

    expect(
      container.querySelector(".Polaris-ResourceList__ResourceListWrapper")
    ).toMatchSnapshot();
  });

  it("renders empty state if no items are passed in", () => {
    const { container } = render(
      <MoviesList
        movies={[]}
        totalResults={0}
        singular="movie"
        plural="movies"
        renderItem={(movie) => <div>{movie.title}</div>}
        emptyStateTitle="Test empty state"
        emptyStateCaption="Test empty state caption"
      />
    );

    expect(
      container.querySelector(".Polaris-ResourceList__ResourceListWrapper")
    ).toMatchSnapshot();
  });
});
