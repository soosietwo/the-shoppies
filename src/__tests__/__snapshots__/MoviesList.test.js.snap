// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList renders and matches snapshot 1`] = `
<div
  class="Polaris-ResourceList__ResourceListWrapper"
>
  <ul
    aria-live="polite"
    class="Polaris-ResourceList"
  >
    <div>
      Test
    </div>
  </ul>
</div>
`;

exports[`MoviesList renders empty state if no items are passed in 1`] = `
<div
  class="Polaris-ResourceList__ResourceListWrapper"
>
  <div
    class="Polaris-Card"
  >
    <div
      class="Polaris-EmptyState Polaris-EmptyState--withinContentContainer"
    >
      <div
        class="Polaris-EmptyState__Section"
      >
        <div
          class="Polaris-EmptyState__DetailsContainer"
        >
          <div
            class="Polaris-EmptyState__Details"
          >
            <div
              class="Polaris-TextContainer"
            >
              <p
                class="Polaris-DisplayText Polaris-DisplayText--sizeSmall"
              >
                Test empty state
              </p>
              <div
                class="Polaris-EmptyState__Content"
              >
                <p>
                  Test empty state caption
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Polaris-EmptyState__ImageContainer"
        >
          <img
            alt=""
            class="Polaris-EmptyState__Image"
            role="presentation"
            src="https://www.iconarchive.com/download/i99782/designbolts/free-multimedia/Film.ico"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
